src/Proto3/Wire/Builder.hs

```diff
@@ -385,10 +390,31 @@ word64LE w = Builder (Sum 8, BB.word64LE w)
 --
 -- This encoding is used in the wire format of Protocol Buffers version 3.
 word64Base128LEVar :: Word64 -> Builder
-word64Base128LEVar i
-    | i <= 0x7f = word8 (fromIntegral i)
-    | otherwise = word8 (fromIntegral (0x80 .|. i)) <>
+
+{-
+word64Base128LEVar !i
+    | i < 0x80 = word8 (fromIntegral i)
+    | otherwise = word8 (fromIntegral i .|. 0x80) <>
           word64Base128LEVar (i `shiftR` 7)
+-}
+
+{-
+
+Prelude Data.Bits> map bit $ take 11 [0,7..]
+[1,128,16384,2097152,268435456,34359738368,4398046511104,562949953421312,72057594037927936,9223372036854775808,1180591620717411303424]
+
+-}
+word64Base128LEVar i =
+    if      i < 128 then word8 (fromIntegral i)
+    else if i < 16384 then Builder (Sum 2) (BB.word8 (fromIntegral i .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 7)))
```

Comment:
Why not `|` notation?

Reply:
good point -- somehow as i was writing it the elses felt they made more sense but they don't